Improve beforeRequest and connectionLabel clarity

bhaity opened the pull request:
### Issue(s)
- The `includeApiKeyHeader` function does not actually add headers; rename it to better reflect this.
- Adds a comment to provide sample code for adding headers to the request (the common `Authorization` header is used as an example)
- Makes sure to pass `bundle` into the `connectionLabel` so that we can access the key from the JSON that is returned

### Bonus
- commits `package-lock.json`

bhaity committed:
be more explicit about adding params vs headers to each request, and ensure the connectionLabel is set correctly based on the example

codebycaleb commented on authentication.js:
```diff
@@ -25,6 +25,8 @@ module.exports = {
+  // assuming "username" is a key in the json returned from testAuth
+  connectionLabel: (z, bundle) => {
+    return bundle.inputData.json.username;
+  }
```
So, taking a look, I definitely agree that there's an error here (i.e. the current code doesn't work and will leave your label as `{{username}}`. 
However, I think there's a better fix for this issue. The problem lies in line 14 of this file – instead of returning the `response`, the `testAuth` function should be returning `response.json`. When that happens, you can do
```
connectionLabel: '{{username}}'
```
or
```
connectionLabel: (z, bundle) => {
  return bundle.inputData.username;
}
```

codebycaleb commented on package-lock.json:
```diff
@@ -0,0 +1,451 @@
+      }
+    }
+  }
+}
```
It looks like none of our other example repositories have this file committed, and though I don't know the precise reasoning behind that, I'd suggest this repository should match the others. :)

codebycaleb reviewed (changes requested):
I've added comments on authentication.js and package-lock.json that I think should be addressed before approving changes. :)

bhaity committed:
address review comments - remove package-lock and return json from the testAuth function

bhaity commented:
@codebycaleb thanks, I addressed your concerns, but it seems like the build failed. This could be due to `zapier test` not working at the moment, it seems that we are getting a 503 from the style check endpoint: 
```bash
➜  zapier test --debug

Validating project locally.
No structural errors found during validation routine.
This project is structurally sound!

Checking app style.
>> POST https://zapier.com/api/platform/cli/style-check
>> {"version":"1.0.0","platformVersion":"5.1.0","authentication":{"type":"custom","fields":[{"key":"apiKey","label":"API Key","required":true,"type":"string"}],"test":"$func$1$t$","connectionLabel":"$func$2$f$"},"beforeRequest":["$func$3$f$"],"afterResponse":[],"resources":{},"triggers":{},"searches":{},"creates":{},"searchOrCreates":{},"zip_file":"raw zip removed in logs"}
<< 503
<<

<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" class="home-minimal-html ">
<head>

Error: "https://zapier.com/api/platform/cli/style-check" returned "503" saying "

    at /Users/sikandar/.nvm/versions/node/v6.10.3/lib/node_modules/zapier-platform-cli/lib/utils/api.js:122:15
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)

```
 
Maybe we just need to rebuild once the service is back up.

xavdid commented:
@bhaity I bumped the build, but it looks like the test is failing anyway. I believe because you're returning `response.json`, but the test is looking for the full response object.

bhaity committed:
update test to better reflect the new functionality

bhaity commented:
@xavdid thanks, i updated the spec to reflect the new functionality

bhaity commented:
@xavdid @codebycaleb can we merge this if all looks good?

xavdid commented:
Cool! it looks good to me. @bhaity can you sign the CLA (in the reviews area) and I can merge it.

bhaity commented:
@xavdid done!

xavdid commented:
Awesome. Thanks for your patience here. Great improvement!
